### ISSUE TITLE
Organize directories in a better way

### ISSUE THREAD

**Issue by zamentur** [YunoHost/issues#1936]
TL;DR : YunoHost doesn't organize very well it's directories, and it's difficult to know what to do exactly

Important: that's completely draft. So a lot of things could be bad ideas.

## Situation in YunoHost 4.3

| Path | Content  | 
|--------|---|
| /home/userX                       | User dir (can be accessed by sftp or nextcloud) |
| /home/borguserX                       | User dir for borg | 
| /home/yunohost.app          | Data directory for each apps | 
| /home/yunohost.backup/archives       | Tar archives dir  | 
| /home/yunohost.backup/tmp       | Temporary dir to collect and organize files before backup  | 
| /home/yunohost.multimedia/userX   | Multimedia dir for user (it allows a user to share files between several apps) | 
| /home/yunohost.multimedia/share   | Multimedia share between all users of a yunohost | 
| /var/cache/yunohost                   | Some cache data and backup conf | 
| /var/lib/mysql              | Mysql db | 
| /var/lib/postgresql         | Postgresql db | 
| /var/mail                   | User mail  | 
| /var/www                    | Web app software  | 
| /opt/appX                        | (obsolete?) App software installed in /opt | 
| /opt/yunohost/appX                        | (obsolete?) An other app softwaredir | 
| /opt/node_n                     | Node dependencies | 
| /opt/depX                        | App dependencies installed in /opt | 
| /var/log                    | Logs (could be in ram) | 

## About multi disk/partition setup
Several yunohost setup used several disk (SD, SSD, HDD, ...). Some disk are cheaped, more fast or more reliable than other and user can want to put data or software specifically on some of them.

See https://github.com/YunoHost/issues/issues/1823

In general: 
- user data should go on HDD (more reliable and less expensive than SSD/SD)
- databases should go on SSD (if you have SD and HDD, it's not easy to decide where put your db files)
- software/dependencies/conf should go on the fastest disk

## An apps directory to rule them all
It's not coherent to have apps software dir in :
- /var/www
- /opt
- /opt/yunohost

Some apps are not webapp, so it could be a good idea to not put them in /var/www.

The ideal could be to have a /apps dir at the root of the system (but migration with multi disk/partition could be complex).

## Bring piece of apps together ?????
This apps dir could bring together all apps part in order to ease user comprehension and to prepare snapshot features ?

Draft of what it could look:
```
/apps
├── borgserver__X553
│   ├── data
│   ├── etc
│   ├── log
│   ├── package
│   │   ├── conf
│   │   ├── hooks
│   │   ├── manifest.json
│   │   ├── scripts
│   │   └── settings.yml
│   └── software
│       ├── app
│       └── lib
├── nextcloud__xbG4
│   ├── data
│   ├── database
│   │   └── mysql
│   │       ├── db.opt
│   │       ├── table1.frm
│   │       ├── table1.ibd
│   │       ├── table2.frm
│   │       └── table2.ibd
│   ├── etc
│   ├── log
│   ├── package
│   │   ├── conf
│   │   ├── hooks
│   │   ├── manifest.json
│   │   ├── scripts
│   │   └── settings.yml
│   └── software
│       ├── app
│       └── lib
├── vpnclient
│   ├── etc
│   ├── log
│   ├── package
│   │   ├── conf
│   │   ├── hooks
│   │   ├── manifest.json
│   │   ├── scripts
│   │   └── settings.yml
│   └── software
│       └── bin
└── wordpress__T7Dz
    ├── data
    ├── database
    │   └── mysql
    │       ├── db.opt
    │       ├── table1.frm
    │       ├── table1.ibd
    │       ├── table2.frm
    │       └── table2.ibd
    ├── etc
    ├── log
    ├── package
    │   ├── conf
    │   ├── hooks
    │   ├── manifest.json
    │   ├── scripts
    │   └── settings.yml
    └── software
        ├── app
        └── lib
```
Some part should be linked in good directory (like /var/lib/mysql/ thanks to mount --bind). <= I am not sure it's a good idea.

## [DONE] /home/yunohost.conf -> /var/?
~~The /home/yunohost.conf seems not to be in the right place, /var seems better to store this kind of small technical files~~~

This dir has been moved to `/var/cache/yunohost/appconfbackup/`

## User multimedia and mail

Could it be a good idea to move this dir into the /home/user dir ?

For example:
```
/home/user1
├── .ssh
├── .mail
├── Document
├── eBook
├── Music
├── Picture
└── Video
```
There could be several way to access home folder:
- ssh
- sftp
- samba
- home mounted in nextcloud
- Using a desktop environment (yunohost+xfce for example), note in this case the desktop env will probably translate dir name :/

## Backup/restore work dir (like /home/yunohost.backup/tmp)

In order to increase restore performance, this dir needs to be on same partition as the biggest datadir on the server.

## Backup archives
The dir /home/yunohost.backup/archives should ideally not be on the same disk than original data.

##  Shared folder feature (like /home/yunohost.multimedia/share)

Share files between all account is interesting on small instances but with bigger communities it's not always a good idea. That's why the /home/yunohost.multimedia/share approach is a bit unadapted

We probably want to be able to segment by group.

And there are the wish to have NAS features like share a file with several methods like samba or sftp.

May be should have a /home/groups folder, like that we could do:
```
/home/groups
├── all_users
│   ├── Document
│   ├── eBook
│   ├── Music
│   ├── Picture
│   └── Video
├── group1
│   ├── Document
│   ├── eBook
│   ├── Music
│   ├── Picture
│   └── Video
└── group2
    ├── Document
    ├── eBook
    ├── Music
    ├── Picture
    └── Video
```
